### PR TITLE
add hash for exponential double values.

### PR DESCRIPTION
**Pull request by vmidyllic**

### CODE REVIEW

**Review by Kolezhniuk — CHANGES_REQUESTED** [review]

File: src/lib/utils.ts [hunk 1]
```diff
@@ -85,18 +85,40 @@ export const convertStringToXsdValue = (dataType: string, valueStr: string): Val
       }
       break;
     }
+    // this is for v2
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply src/lib/utils.ts:88]: why. it's for v2

File: src/lib/types/types.ts [hunk 2]
```diff
@@ -20,8 +20,12 @@ export enum XSDNS {
   NonPositiveInteger = 'http://www.w3.org/2001/XMLSchema#nonPositiveInteger',
   NegativeInteger = 'http://www.w3.org/2001/XMLSchema#negativeInteger',
   PositiveInteger = 'http://www.w3.org/2001/XMLSchema#positiveInteger',
-  DateTime = 'http://www.w3.org/2001/XMLSchema#dateTime'
+  DateTime = 'http://www.w3.org/2001/XMLSchema#dateTime',
+  Double = 'http://www.w3.org/2001/XMLSchema#double'
 }
+export const isDouble = (v: number) => String(v).indexOf('.') !== -1 || Math.abs(v) >= 1e21;
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply src/lib/types/types.ts:26]: see nothing wrong with indexof

File: src/lib/utils.ts [hunk 3]
```diff
@@ -85,18 +85,40 @@ export const convertStringToXsdValue = (dataType: string, valueStr: string): Val
...
+    //     value = canonicalDouble(parseFloat(valueStr));
+
     default:
       value = valueStr;
   }
   return value;
 };
 
-export const convertAnyToString = (v: unknown): string => {
+export const convertAnyToString = (v: unknown, datatype:string): string => {
```
[the remark this answers is not in the document]
Comment (Kolezhniuk) [reply src/lib/utils.ts:98]: but in this case, you're checking `typeof v` two times, why do we need that if we can check it once

**Review by Kolezhniuk — CHANGES_REQUESTED** [review]

File: src/lib/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kolezhniuk) [reply src/lib/utils.ts]: than it's better to mark this comment with TODO

**Review by Kolezhniuk — APPROVED** [review]

**Pull request merged**